AccessYouth/AccessYouthOperator/ViewController.swift

```diff
@@ -7,14 +7,87 @@
 //
 
 import UIKit
+import MapKit
+import CoreLocation
+import Foundation
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, CLLocationManagerDelegate {
+    let locationManager = CLLocationManager()
+    let broadcastButton = UIButton()
+    var mapView: MKMapView?
+    var timer: Timer?
+    var busLocation = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+    var isBroadcastOn = false
+    let broadcastInterval: Double = 1.0
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        // set up mapview
+        self.mapView = MKMapView(frame: CGRect(x: 0, y: 0, width: view.frame.size.width,
+                                               height: view.frame.size.height))
+        if let mapView = self.mapView {
+            mapView.mapType = MKMapType.standard
+            mapView.isZoomEnabled = true
+            mapView.isScrollEnabled = true
+            mapView.showsUserLocation = true
+            let region = MKCoordinateRegion(center: busLocation, span:
+                MKCoordinateSpan(latitudeDelta: 0.02, longitudeDelta: 0.02))
+            mapView.setRegion(region, animated: true)
+            self.view.addSubview(mapView)
+        }
+
+        // set up button to turn on/off location broadcast of bus location
+        broadcastButtonSetup()
+
+        // request location access
+        locationManager.requestAlwaysAuthorization()
+        locationManager.requestWhenInUseAuthorization()
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.delegate = self
+            locationManager.desiredAccuracy = kCLLocationAccuracyBestForNavigation
+            locationManager.startUpdatingLocation()
+        }
+
+        // set up timer to poll for location at a fixed interval
+        timer = Timer.scheduledTimer(timeInterval: broadcastInterval, target: self, selector: #selector(sendLocation),
+                                     userInfo: nil, repeats: true)
     }
 
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        if let location = locations.last {
+            let coordinate = location.coordinate
+            busLocation = CLLocationCoordinate2D(latitude: coordinate.latitude, longitude: coordinate.longitude)
```

Comment:
nit: location.coordinate is already of type CLLocationCoordinate2D, so you just have to do `busLocation = location.coordinate`